projects/fal/src/fal/api.py

```diff
@@ -754,10 +758,20 @@ def __call__(self, *args, **kwargs) -> None:
                 f"[warning] {self._func.__name__} function is served with no arguments."
             )
 
-        from uvicorn import run
+        from fastapi import FastAPI
+        from starlette_exporter import PrometheusMiddleware, handle_metrics
+        from uvicorn import Config, Server
 
         app = self.build_app()
-        run(app, host="0.0.0.0", port=8080)
+        app.add_middleware(PrometheusMiddleware, prefix="http")
+        metrics_app = FastAPI()
```

Comment:
we currently have two implementations of this, one here and one in `fal/app.py`, @squat can you address both of these (we should consolidate them into a single implementation sometime soon).

Reply:
ahh interesting. that would be very nice to consolidate :heart: thanks for the pointer